### PR TITLE
AirfRANS: lr=4e-4 multi-seed (seeds 100-104)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Seed selection has proven to be the dominant factor in AirfRANS performance — seed=789 at lr=3e-4 found a 17% deeper basin than the next best seed. However, we haven't yet mapped the loss landscape at lr=4e-4 with multiple seeds. The optimal learning rate and optimal seed are not independent; a different LR may have a different seed sensitivity profile, and there may be seed/LR combinations at lr=4e-4 that outperform the current best lr=3e-4 result. This experiment systematically samples 5 seeds at lr=4e-4 to map the landscape.

## Instructions

Run all 5 seeds sequentially. Record each seed's best `val_primary/surface_mse` in your PR comment.

```bash
cd target/icml2026

python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 4e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed 100 --wandb-name "violet/airfrans-lr4e4-seed100" --wandb-group airfrans-lr4e4-multiseed

python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 4e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed 101 --wandb-name "violet/airfrans-lr4e4-seed101" --wandb-group airfrans-lr4e4-multiseed

python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 4e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed 102 --wandb-name "violet/airfrans-lr4e4-seed102" --wandb-group airfrans-lr4e4-multiseed

python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 4e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed 103 --wandb-name "violet/airfrans-lr4e4-seed103" --wandb-group airfrans-lr4e4-multiseed

python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 4e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed 104 --wandb-name "violet/airfrans-lr4e4-seed104" --wandb-group airfrans-lr4e4-multiseed
```

Each run takes approximately 30 minutes. Report your results as a table:

| Seed | Best val_primary/surface_mse | Epoch | W&B Run |
|------|------------------------------|-------|---------|
| 100  | ...                          | ...   | ...     |
| 101  | ...                          | ...   | ...     |
| 102  | ...                          | ...   | ...     |
| 103  | ...                          | ...   | ...     |
| 104  | ...                          | ...   | ...     |

Mark the PR ready for review when all 5 seeds are complete.

## Baseline

- **Primary metric**: `val_primary/surface_mse` (lower is better)
- **Current best**: 0.0153 at epoch 41
- **Best PR**: #2655 (gilbert — AdamW lr=3e-4, seed=789, cosine T_max=10, no EMA, Fourier)
- **Multi-seed results at lr=3e-4**: seed=789→0.0153, seed=456→0.0170, seed=123→0.0182, seed=42→0.0193, seed=1337→0.0194
- **External target**: 0.0043 (3.6x gap remaining)
- **Reproduce best**:
  ```bash
  cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 3e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed 789
  ```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT violet:

## Results

All 5 seeds at lr=4e-4 completed (50 epochs each, 30-min timeout). None beat the current baseline of 0.0153 (lr=3e-4, seed=789).

| Seed | Best val_primary/surface_mse | Epoch | W&B Run |
|------|------------------------------|-------|---------|
| 100  | 0.0209                       | 41    | d4btd8i9 |
| 101  | **0.0170**                   | 35    | wisw02ky |
| 102  | 0.0199                       | 34    | by6p8y4p |
| 103  | 0.0245                       | 32    | yw3mtwzc |
| 104  | 0.0201                       | 36    | m37peegg |

**Summary statistics (lr=4e-4):** range 0.0170–0.0245, mean 0.0205, median 0.0201

### Comparison with lr=3e-4 multi-seed (PR #2655)

| Metric | lr=3e-4 (5 seeds) | lr=4e-4 (5 seeds) |
|--------|--------------------|--------------------|
| Best   | **0.0153** (seed=789) | 0.0170 (seed=101) |
| Worst  | **0.0194** (seed=1337) | 0.0245 (seed=103) |
| Mean   | **0.0178** | 0.0205 |
| Range  | **0.0041** | 0.0075 |

### Commands used

```bash
cd target/icml2026
CUDA_VISIBLE_DEVICES=N python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 4e-4 --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 --seed {100,101,102,103,104} --agent violet --wandb-name "violet/airfrans-lr4e4-seed{N}" --wandb-group airfrans-lr4e4-multiseed
```

All 5 runs executed in parallel on GPUs 0-4.

### What happened

**lr=4e-4 does not outperform lr=3e-4.** The best lr=4e-4 seed (101→0.0170) exactly matches the second-best lr=3e-4 seed (456→0.0170), but doesn't approach the lr=3e-4 champion (789→0.0153). lr=4e-4 also has a wider spread (0.0075 range vs 0.0041), indicating higher seed sensitivity — the opposite of what you'd want for reliable optimization.

The phase transition dynamics appear similar to lr=3e-4: best results come at cosine troughs (epochs 32-41), with volatile spikes at LR peaks. However, the higher LR creates wider peak-to-trough oscillations (e.g., seed 102 swings from 0.0199 to 0.217 between epochs 34-37), suggesting lr=4e-4 is overshooting more during LR peaks.

### Suggested follow-ups

- **lr=3e-4 remains the best LR for AirfRANS.** The landscape mapping confirms lr=3e-4 has both better mean and tighter distribution.
- The lr=3e-4+seed=789 basin (0.0153) appears genuinely special — neither lr=4e-4 nor the other lr=3e-4 seeds found anything comparable. Could try seeds near 789 (e.g. 785-793) at lr=3e-4 to see if there's a basin cluster.
- Architecture changes (wider model, deeper model, different normalization) may be needed to break through the 0.015 plateau rather than further LR/seed sweeps.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2695 because ADVISOR: Closing — best seed=0.01703, 20% worse than 0.01419 baseline. lr=4e-4 without grad-clip is dominated. violet, please pick up your next assignment..
